Add prototype of Google AdMob ads

nilsreichardt opened the pull request:
## Description

This pull request adds Google AdMob for displaying ads in Sharezone. It's an experiment that we will run in the next weeks. Because we set the [`tagForUnderAgeOfConsent`](https://developers.google.com/admob/flutter/privacy/gdpr#tag_for_under_age_of_consent) to `true`, we don't need to show a consent dialog because only ["limited ads"](https://developers.google.com/admob/flutter/privacy/ad-serving-modes#limited_ads) will be displayed. 

Use the activation code "ads" to enable ads.

## Demo

| Dashboard | Timetable | Homework | Info Screen |
|--------|--------|-----|---------|
| ![Screenshot_20241012-221743](https://github.com/user-attachments/assets/280dadcf-4272-4507-b99c-fe31a7093521) | ![Screenshot_20241012-222240](https://github.com/user-attachments/assets/744fba56-aa97-49e0-9c13-deaf5873dadf) | ![Screenshot_20241012-222232](https://github.com/user-attachments/assets/e812351b-22f9-4f0e-b682-9625d13e3d82) | ![Screenshot_20241013-184158](https://github.com/user-attachments/assets/6920ff44-0f2f-4b38-91f2-538625bf57b4) |

Additionally, we show after every fifth homework check a full-screen ad:

https://github.com/user-attachments/assets/62ca4b3f-c30a-4835-b10d-678fddf48f1a

Sharezone-Admin commented:
⬇️ Builds generated by [Codemagic](https://codemagic.io) for commit e4e4af15e4a43aea0040144141720ca309c623ec ⬇️

_Note_: Only Sharezone team members are able to install the iOS app.

| iOS | | Android |
|:-:|:-:|:-:|
| ![iOS QR code](https://app-preview-qr.nils.re/?size=150x150&data=https%3A%2F%2Fapi.codemagic.io%2Fartifacts%2F.eJwVwUuOgjAAANC7uCdRKq0sXBQoVSvyZ8CNwU-oOhQMFUpPP5n3Fg3-5wgvGGcRxcdrFZq0sjps4UJyqHwSTipIBGejEokEr-dEp1iduuxwAPrck6PB3DN2ytgXoSH85y-NX8s1MlNzQ0mBzeWnhNZw7RiJI7ACc9BWD6lUTXN_Xt_NtSpKt7HOEXIdN_7uNj6zufmwW_a-VG0H4XzhiWdMHrfrQd81ZwhSQ9g_Te4CMDi3aAU9jeBOsvR96iUmF5Ek7NYQG45h3T1Tie7fYc_RshdZ5nn6Gz2wkKbG7RCgEgNaV3l32GdxZWyu1lHBXO4LHPRkJEKfPtN2u_gDLtdgMQ.jj9TyLmj3v5s9AchQReBLaUXVpU&platform=ios&groupId=a2ba0510) <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVwUuOgjAAANC7uCdRKq0sXBQoVSvyZ8CNwU-oOhQMFUpPP5n3Fg3-5wgvGGcRxcdrFZq0sjps4UJyqHwSTipIBGejEokEr-dEp1iduuxwAPrck6PB3DN2ytgXoSH85y-NX8s1MlNzQ0mBzeWnhNZw7RiJI7ACc9BWD6lUTXN_Xt_NtSpKt7HOEXIdN_7uNj6zufmwW_a-VG0H4XzhiWdMHrfrQd81ZwhSQ9g_Te4CMDi3aAU9jeBOsvR96iUmF5Ek7NYQG45h3T1Tie7fYc_RshdZ5nn6Gz2wkKbG7RCgEgNaV3l32GdxZWyu1lHBXO4LHPRkJEKfPtN2u_gDLtdgMQ.jj9TyLmj3v5s9AchQReBLaUXVpU) | | ![Android QR code](https://app-preview-qr.nils.re/?size=150x150&data=https%3A%2F%2Fapi.codemagic.io%2Fartifacts%2F.eJwVwUuOgjAAANC7uCeRTxEWLmgFjbbWKjjVDZmCBZFfQRA8_WTeW2TeP1hvyDgPono-ZLE7v9mt3Hnp3sBbX5CTLvU6aWXwfqE8V4kMKG5mkmv8tjylaYat-NL88vgLDrZ0DU01Sc6z4hCNKeqdz94cgJ8G6qMApmcAXW3qukNbo-8VvpjXYVyKsM9UZSwBtcSz6GzG737pkNUxjDkbIzRBcllaWt-U0KVY3nzZqLt4IJCc4Ku1VupOPO0ZfYauCJ1ota9aaBqByWHkisl-88TFPKtHLretk7FeZ2UYF6rpTDDXlOZX4-HoR2H39nX2KzEOE9qiDQ1K8kNNtl4v_gAS52JU._qMJuu9M55vDQTjagIIXyCIoHds&platform=android&groupId=a2ba0510) <br /> [Download link](https://api.codemagic.io/artifacts/.eJwVwUuOgjAAANC7uCeRTxEWLmgFjbbWKjjVDZmCBZFfQRA8_WTeW2TeP1hvyDgPono-ZLE7v9mt3Hnp3sBbX5CTLvU6aWXwfqE8V4kMKG5mkmv8tjylaYat-NL88vgLDrZ0DU01Sc6z4hCNKeqdz94cgJ8G6qMApmcAXW3qukNbo-8VvpjXYVyKsM9UZSwBtcSz6GzG737pkNUxjDkbIzRBcllaWt-U0KVY3nzZqLt4IJCc4Ku1VupOPO0ZfYauCJ1ota9aaBqByWHkisl-88TFPKtHLretk7FeZ2UYF6rpTDDXlOZX4-HoR2H39nX2KzEOE9qiDQ1K8kNNtl4v_gAS52JU._qMJuu9M55vDQTjagIIXyCIoHds) |

<sub>(expires Wed, 13 Nov 2024 07:44:50 GMT)<br/>Codemagic build: [670cc8a2589de75c791610db](https://codemagic.io/app/629c82ea463af7ff553fc7a5/build/670cc8a2589de75c791610db); comment generated with [Codemagic App Preview](https://github.com/nilsreichardt/codemagic-app-preview)</sub>

<!-- Codemagic App Preview; appName: default -->

Jonas-Sander commented:
For me the ads are not activated after using the code "ads" from the apk of the app-preview.

Jonas-Sander commented:
> Additionally, we show after every fifth editing/adding homework a full-screen ad:

I thought we wanted to show it after completing a homework?

Jonas-Sander commented on app/lib/ads/ad_info_dialog.dart:
```diff
@@ -0,0 +1,56 @@
+            children: [
+              const TextSpan(
+                text:
+                    'Wir führen ein Experiment mit Werbung in Sharezone durch. Wenn du keine Werbung sehen möchten, kannst du ',
```
We could commit to a certain time period (e.g. 1-2 months) for the experiment and mention it here so that it might cause less negative reactions (just an idea)

nilsreichardt replied:
Something like:

```suggestion
                    'In den nächsten Wochen führen ein Experiment mit Werbung in Sharezone durch. Wenn du keine Werbung sehen möchten, kannst du ',
```

Or more specific with the timeframe?

Jonas-Sander replied:
Yes thats fine, but I'd say: "_Innerhalb_ der nächsten Wochen [...]"

nilsreichardt commented:
> I thought we wanted to show it after completing a homework? I'm not sure I like it after editing/adding a homework.

Yes, I just added to editing/adding a homework because I thought adding it to completing a homework would be harder to add. But it wasn't that complicated - so I replaced it as we discussed 👍

Jonas-Sander reviewed (approved):
They are still not shown on my device, but LGTM
